Create archiver directory automatically if it's missing

An exception is thrown if the archiver directory provided for the server does not exist:
`java.lang.RuntimeException: java.io.IOException: Archiver directory doesn't exist: /nail/home/sarthakn/nrtsearch_talk/archiver
        at org.apache.platypus.server.luceneserver.StartIndexHandler.downloadArtifact(StartIndexHandler.java:133)
        at org.apache.platypus.server.grpc.LuceneServer$LuceneServerImpl.startIndex(LuceneServer.java:302)
        at org.apache.platypus.server.grpc.LuceneServerGrpc$MethodHandlers.invoke(LuceneServerGrpc.java:2109)
        at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:172)
        at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)
        at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:817)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: java.io.IOException: Archiver directory doesn't exist: /nail/home/sarthakn/nrtsearch_talk/archiver
        at org.apache.platypus.server.utils.ArchiverImpl.download(ArchiverImpl.java:92)
        at org.apache.platypus.server.luceneserver.StartIndexHandler.downloadArtifact(StartIndexHandler.java:131)
        ... 12 more`

We can just create the directory if it's missing instead of throwing an exception.

--- a/src/main/java/com/yelp/nrtsearch/server/utils/ArchiverImpl.java
+++ b/src/main/java/com/yelp/nrtsearch/server/utils/ArchiverImpl.java
@@ -89,7 +89,8 @@ public class ArchiverImpl implements Archiver {
     @Override
     public Path download(String serviceName, String resource) throws IOException {
         if (!Files.exists(archiverDirectory)) {
-            throw new IOException("Archiver directory doesn't exist: " + archiverDirectory);
+            logger.info("Archiver directory doesn't exist: " + archiverDirectory +  " creating new ");
+            Files.createDirectory(archiverDirectory);
         }
 
         final String latestVersion = getVersionString(serviceName, resource, "_latest_version");

